lib/SIL/SILVerifier.cpp

```diff
@@ -619,20 +623,32 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
   }
 
+  static unsigned numInstsInFunction(const SILFunction &F) {
+    unsigned numInsts = 0;
+    for (auto &BB : F) {
+      numInsts += std::distance(BB.begin(), BB.end());
+    }
+    return numInsts;
+  }
+
   SILVerifier(const SILFunction &F, bool SingleFunction = true)
       : M(F.getModule().getSwiftModule()), F(F),
         fnConv(F.getLoweredFunctionType(), F.getModule()),
         TC(F.getModule().Types), OpenedArchetypes(&F), Dominance(nullptr),
+        InstNumbers(numInstsInFunction(F)),
```

Comment:
@eeckstein InstNumbers is a DenseMap. Why are you doing doing an O(n) traversal to avoid log(N) hashtable growth?

Reply:
For the record, it's just done to minimize allocation on the common small function case and won't increase overall complexity for large functions.